README.md

```diff
@@ -1,4 +1,253 @@
+  pointing to the CSV files you want to use to define your permissions
+
+#### Autoreload
+
+`csv_permissions` caches the data read from the CSV permissions file on server start.
+During development this means you have to manually restart the dev server when you make changes.
+You can hook into django's autoreloader to automatically reload when the CSV file is changed:
+
+In one of your [django app configs](https://docs.djangoproject.com/en/dev/ref/applications/#for-application-authors):
+
+```python
+def add_csv_permissions_watcher(sender, **kwargs):
+    """In dev we want to reload if the csv permission file changes"""
+    sender.extra_files.add(settings.CSV_PERMISSIONS_PATH)
+
+class MySiteAppConfig(AppConfig):
+    name = "my_site"
+    verbose_name = "My Site"
+
+    def ready(self):
+        if settings.DEBUG:
+            from django.utils.autoreload import autoreload_started
+
+            autoreload_started.connect(add_csv_permissions_watcher)
+```
+
+If you're using [`runserver_plus`](https://django-extensions.readthedocs.io/en/latest/runserver_plus.html)
+from `django-extensions` you can add your CSV files to
+[`settings.RUNSERVER_PLUS_EXTRA_FILES`](https://django-extensions.readthedocs.io/en/latest/runserver_plus.html#configuration).   
+
+  
+### The CSV File
+
+An example permission file:
+
+```csv
+Model,     App,           Action,            Is Global, admin, assistant, customer
+
+# Comment lines and blank lines will be ignored
+
+Publisher, library,       add,               yes,       yes,
+Publisher, library,       view,              no,        all,
+Publisher, library,       change,            no,        all,
+Publisher, library,       delete,            no,        all,
+
+Book,      library,       add,               yes,       yes,   yes,
+Book,      library,       view,              no,        all,   all,
+Book,      library,       change,            no,        all,   all,
+Book,      library,       delete,            no,        all,   all,
+
+Loan,      library,       add,               yes,       yes,   yes,       yes,
+Loan,      library,       view,              no,        all,   all,       own,
+Loan,      library,       change,            no,        all,   all,
+Loan,      library,       delete,            no,        all,   all,
+
+# The model column can be blank:
+
+,          library,       report_outstanding,yes,      yes,   yes,
+,          library,       report_popularity, yes,      yes,   yes,
+```
+
+The first 4 columns define the permission details:
+
+**Model** is used to resolve the permission name but is otherwise not used. There is no checks that objects passed to the `has_perm()` actually match the correct type.
+
+**App** is used to resolve the permission name and model.
+
+**Action** is an arbitrary identifier that is used to resolve the permission name.
+
+**Is Global** whether the permission is global or per-object (see "Global Permission" section below)
```

Comment:
it looks like we validate that a permission must be global if Model is not defined. 2 things

1) Is this a hard requirement? could you have a case where you want a custom non-model permission that does accept an object of some kind?

2) if it is a hard requirement should mention it here somewhere

Reply:
> could you have a case where you want a custom non-model permission that does accept an object of some kind?

I hadn't thought of that; conceivably yes. Have added a comment that this restriction may be relaxed in future.